webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/TableRow.js

```diff
@@ -0,0 +1,289 @@
+  FlagIcon,
+} from '@patternfly/react-icons';
+import {
+  SelectVariant,
+  Select,
+  SelectOption,
+  Button,
+  TextInput,
+  TextArea,
+  Checkbox,
+  FormGroup,
+  Form,
+  Tooltip,
+} from '@patternfly/react-core';
+import { APIActions } from '../../../../redux/API';
+import { sprintf, translate as __ } from '../../../../common/I18n';
+import { HOST_PARAM, columnNames, typeOptions } from './ParametersConstants';
+import { updateHost } from '../../ActionsBar/actions';
+import { RowActions } from './RowActions';
+
+const getValue = param => {
+  if (param['hidden_value?']) {
+    return '••••••••';
+  }
+  if (param.parameter_type === 'boolean') {
+    return param.value.toString();
+  }
+  if (!param.value)
+    return <span className="disabled-text">{__('No value')}</span>;
+  if (['json', 'yaml', 'array', 'hash'].includes(param.parameter_type)) {
+    return JSON.stringify(param.value);
+  }
+  return param.value;
+};
+
+export const ParametersTableRow = ({
+  param,
+  rowIndex,
+  editingRow,
+  setEditingRow,
+  hostId,
+  editHostsPermission,
+  isNew,
+}) => {
+  const [name, setName] = useState(param.name);
+  const [type, setType] = useState(param.parameter_type);
+  const [value, setValue] = useState(param.value);
+  const [selectIsOpen, setSelectIsOpen] = useState(false);
+  const [selectValueIsOpen, setSelectValueIsOpen] = useState(false);
+  const [isHide, setIsHide] = useState(param['hidden_value?']);
+
+  const dispatch = useDispatch();
+
+  const APIActionsProps = {
+    errorToast: ({ response }) =>
+      // eslint-disable-next-line camelcase
+      response?.data?.error?.full_messages || response?.data?.error?.message,
+    handleSuccess: () => {
+      dispatch(updateHost(hostId));
+    },
+  };
+  const onSubmit = () => {
+    if (isNew || param.associated_type !== HOST_PARAM) {
+      dispatch(
+        APIActions.post({
+          url: `/api/hosts/${hostId}/parameters/`,
+          key: 'NEW-PARAM',
+          params: {
+            name,
+            value,
+            parameter_type: type,
+            hidden_value: isHide,
+          },
+          successToast: () =>
+            sprintf(__('Parameter %s has been created successfully'), name),
+          ...APIActionsProps,
+        })
+      );
+    } else {
+      dispatch(
+        APIActions.put({
+          url: `/api/hosts/${hostId}/parameters/${param.id}`,
+          key: `${param.id}-EDIT`,
+          params: {
+            name,
+            value,
+            parameter_type: type,
+            hidden_value: isHide,
+          },
+          successToast: () =>
+            sprintf(__('Parameter %s has been edited successfully'), name),
+          ...APIActionsProps,
+        })
+      );
+    }
+  };
+  const onCancel = () => {
+    setName(param.name);
+    setType(param.parameter_type);
+    setValue(param.value);
+    setIsHide(false);
+    setSelectIsOpen(false);
+  };
+  const isEditing = isNew || editingRow === rowIndex;
+  return (
```

Comment:
This is three giant ternaries. Would it be more readable to just make different components for editing / not editing?

(not a blocker; I'll leave this one up to you)